lib/hpke/src/certificate.cpp

```diff
@@ -80,6 +86,45 @@ struct Certificate::ParsedCertificate
     return digest;
   }
 
+  // Parse Subject fields (CN, O=)
+  static ParsedX509Subject parse_subject(X509* cert)
```

Comment:
This should work on an `const X509_NAME*` so that you can use it with both subject and issuer.

Reply:
Also, please add a comment here noting that this method does not implement fully general name parsing.  Duplicate entries are not supported; if they are present, the last one presented by OpenSSL is chosen.